pkg/resource/nodegroup/hook.go

```diff
@@ -77,6 +79,27 @@ func customPreCompare(
 	if a.ko.Spec.Tags == nil && b.ko.Spec.Tags != nil {
 		a.ko.Spec.Tags = map[string]*string{}
 	}
+	if a.ko.Spec.Taints != nil && a.ko.Spec.Taints == nil || a.ko.Spec.Taints == nil && a.ko.Spec.Taints != nil {
+		delta.Add("Spec.Taints", a.ko.Spec.Taints, b.ko.Spec.Taints)
+	} else if a.ko.Spec.Taints != nil {
+		if len(a.ko.Spec.Taints) != len(b.ko.Spec.Taints) {
+			delta.Add("Spec.Taints", a.ko.Spec.Taints, b.ko.Spec.Taints)
+		} else {
+			for _, taintA := range a.ko.Spec.Taints {
+				var matched = false
+				for _, taintB := range b.ko.Spec.Taints {
+					if reflect.DeepEqual(taintA, taintB) {
+						matched = true
+						break
+					}
+				}
+				if !matched {
+					delta.Add("Spec.Taints", a.ko.Spec.Taints, b.ko.Spec.Taints)
+					break
+				}
+			}
```

Comment:
how about the taints that are in B, but not in A?

Reply:
Just above I am checking that length are equals.